### PR TITLE
Support maskinporten service account tokens

### PR DESCRIPTION
**Pull request by kschulst**

### CODE REVIEW

File: src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java [hunk 1]
```diff
@@ -1,31 +1,40 @@
-package no.ssb.maskinporten.guardian.config;
+package no.ssb.guardian.maskinporten.config;
 
 import io.micronaut.context.annotation.EachProperty;
 import io.micronaut.context.annotation.Parameter;
 import lombok.Data;
 
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
 import java.util.HashSet;
```
Comment (fenglin-ssb) [inline src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java:9]: import javax.validation.constraints.Pattern;

File: src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
 public class MaskinportenClientConfig {
 
     /** clientId is the maskinporten client ID from samarbeidsportalen, typically a UUID */
+    @NotNull
     private String clientId;
 
     /** description is an optional string that describes some contextual information about the maskinporten client. Only used for documentation purposes. */
     private String description;
 
+    // TODO: Validate ending slash using bean validation
```
Comment (fenglin-ssb) [inline src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java:24]: @Pattern(
            regexp = ConfigValidator.AUDIENCE_VALIDATOR,
            message = "Invalid audience: " + ConfigValidator.INVALID_AUDIENCE_MESSAGE
    )

File: src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java [hunk 3]
```diff
@@ -1,31 +1,40 @@
-package no.ssb.maskinporten.guardian.config;
+package no.ssb.guardian.maskinporten.config;
```
Comment (fenglin-ssb) [inline src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java:2]: import no.ssb.guardian.maskinporten.util.ConfigValidator;

package no.ssb.guardian.maskinporten.util;


import java.util.regex.Pattern;

public class ConfigValidator {

    public static final String INVALID_AUDIENCE_MESSAGE = "Audiance must be ended with a backslash (/)";
    public static final String AUDIENCE_VALIDATOR = ".*([.,\/])"; // better regex pattern?
    private static final Pattern AUDIENCE_PATTERN = Pattern.compile(AUDIENCE_VALIDATOR);

    public static boolean isValidAudience(String name) {
        return AUDIENCE_PATTERN.matcher(name).matches();
    }
}

**Review by fenglin-ssb — CHANGES_REQUESTED** [review]
really a lot of work, continue tomorrow. is the bean validator ok?

File: src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java (hunk 1 above)
Comment (kschulst) [reply src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java:9]: Fixed

File: src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java (hunk 2 above)
Comment (kschulst) [reply src/main/java/no/ssb/guardian/maskinporten/config/MaskinportenClientConfig.java:24]: See comment regarding this below

**Pull request merged**